pallets/interest-accrual/src/lib.rs

```diff
@@ -370,40 +387,52 @@ pub mod pallet {
 		}
 
 		pub fn reference_interest_rate(interest_rate_per_sec: T::InterestRate) -> DispatchResult {
-			Rate::<T>::mutate(interest_rate_per_sec, |rate| {
-				if let Some(rate) = rate {
-					rate.reference_count += 1;
-				} else {
-					let rate_count = RateCount::<T>::get();
-					ensure!(
-						rate_count < T::MaxRateCount::get(),
-						Error::<T>::TooManyRates
-					);
-
-					*rate = Some(RateDetailsOf::<T> {
-						accumulated_rate: One::one(),
-						reference_count: 1,
-					});
-					RateCount::<T>::set(rate_count.checked_add(1).ok_or(Error::<T>::TooManyRates)?);
+			Rates::<T>::try_mutate(|rates| {
+				for rate in rates.iter_mut() {
+					if rate.interest_rate_per_sec == interest_rate_per_sec {
+						rate.reference_count += 1;
```

Comment:
Should: Use `ensure_add_assign`

```suggestion
						rate.reference_count.ensure_add_assign(1)?;
```